File: sol/solution/0101-0200/0107/Solution.php
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     public $val = null;
 *     public $left = null;
 *     public $right = null;
 *     function __construct($val = 0, $left = null, $right = null) {
 *         $this->val = $val;
 *         $this->left = $left;
 *         $this->right = $right;
 *     }
 * }
 */
class Solution {
    /**
     * @param TreeNode $root
     * @return Integer[][]
     */
    function levelOrderBottom($root) {
        $result = [];
        if (!$root) return $result;
        $queue = [$root];
        while (!empty($queue)) {
            $levelSize = count($queue);
            $level = [];
            for ($i = 0; $i < $levelSize; $i++) {
                $node = array_shift($queue);
                $level[] = $node->val;
                if ($node->left) $queue[] = $node->left;
                if ($node->right) $queue[] = $node->right;
            }
            $result[] = $level;
        }
        return array_reverse($result);
    }
}

File: sol/solution/0101-0200/0107/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        List<List<Integer>> ans = new LinkedList<>();
        bfs(root, ans, 0);
        return ans;
    }
     public void bfs(TreeNode root, List<List<Integer>>ans, int level) {
        if(root == null) return;
        if(ans.size() == level) {
        ans.add(0, new ArrayList<>());
        }
        ans.get(ans.size() -1- level).add(root.val);
        bfs(root.left, ans, level+1);
        bfs(root.right, ans, level+1);
    }
}